### PR TITLE
Data center url support in Service constructor

### PR DESCRIPTION
**Pull request by jrebs**
The `$config['host']` value is now evaluated during service construction and if the legacy generic web services URL is seen (`https://webservices.netsuite.com`), then the constructor will fetch the appropriate data center url for the configured account.

### CODE REVIEW

**Comment by nastystyles** [comment]
thanks for this.

**Comment by vinu2003** [comment]
Hi, I am trying to use your Library and waiting for this fix to be merged.
Basically you must have retained the setHost() added in SDK 2019.1 which could have been used after retrieving the data centre details for user credentials login.

File: src/NetSuiteClient.php [hunk 1]
```diff
@@ -48,6 +49,28 @@ public function __construct($config, $options = array(), $client = null)
...
+    }
+
+    /**
+     * Set the data center URL for the configured NetSuite account
+     *
+     * @param array $config
+     *
+     * @return void
+     */
+    public function setDataCenterUrl(array $config): void
```
Comment (vinu2003) [inline src/NetSuiteClient.php]: void return types are only for PHP 7.1. Do we have to use this here ?

Comment (jrebs) [reply src/NetSuiteClient.php]: Good point, Vinu. I am so accustomed to being able to use current version php at all times that I failed to consider that one. I'll fix it.

Comment (vinu2003) [reply src/NetSuiteClient.php]: Awesome @jrebs. this helped us a lot.

**Pull request merged**